src/rpp/rpp/operators/combine_latest.hpp

```diff
@@ -13,59 +13,36 @@
 #include <rpp/operators/fwd.hpp>
 
 #include <rpp/defs.hpp>
-#include <rpp/disposables/composite_disposable.hpp>
+#include <rpp/operators/details/combining_strategy.hpp>
 #include <rpp/operators/details/strategy.hpp>
-#include <rpp/operators/details/utils.hpp>
-#include <rpp/schedulers/current_thread.hpp>
-
-#include <memory>
 
 namespace rpp::operators::details
 {
 template<rpp::constraint::observer Observer, typename TSelector, rpp::constraint::decayed_type... Args>
-class combine_latest_disposable final : public composite_disposable
+class combine_latest_disposable final : public combining_disposable<Observer, Args...>
 {
 public:
     explicit combine_latest_disposable(Observer&& observer, const TSelector& selector)
-        : observer_with_mutex{std::move(observer)}
-        , selector{selector}
+        : combining_disposable<Observer, Args...>(std::forward<Observer>(observer))
+        , m_selector(selector)
     {
     }
 
-    pointer_under_lock<Observer> get_observer_under_lock() { return pointer_under_lock{observer_with_mutex}; }
-
-    rpp::utils::tuple<std::optional<Args>...>& get_values() { return values; }
+    const auto& get_selector() const { return m_selector; }
 
-    const TSelector& get_selector() const { return selector; }
-
-    bool decrement_on_completed()
-    {
-        // just need atomicity, not guarding anything
-        return m_on_completed_needed.fetch_sub(1, std::memory_order::seq_cst) == 1;
-    }
+    auto& get_values() { return m_values; }
 
 private:
-    value_with_mutex<Observer>                observer_with_mutex{};
-    rpp::utils::tuple<std::optional<Args>...> values{};
-    RPP_NO_UNIQUE_ADDRESS TSelector           selector;
+    RPP_NO_UNIQUE_ADDRESS TSelector m_selector;
```

Comment:
Most probably crash on MSVC happens due to this line. Try to remove `RPP_NO_UNIQUE_ADDRESS` macro or place this member not first one